src/ducks/subject-viewer.js

```diff
@@ -28,21 +40,55 @@ const subjectViewerReducer = (state = initialState, action) => {
       }
 
       return Object.assign({}, state, {
+        contrast: false,
         rotation: 0,
         scaling: bestFitScale,
         translationX: 0,
         translationY: 0
       });
+    }
+
+    case TOGGLE_CONTRAST: {
+      return Object.assign({}, state, {
+        contrast: !state.contrast
+      });
+    }
+
+    case SET_ROTATION: {
+      let newAngle = action.angle;
+      while (newAngle < 0) { newAngle += 360; }
+      newAngle %= 360;
+
+      return Object.assign({}, state, {
+        rotation: newAngle
+      });
+    }
+
+    case SET_SCALING: {
+      let newScale = (action.scale) ? action.scale : state.scaling;
+      newScale = Math.max(MIN_SCALING, Math.min(MAX_SCALING, newScale));
+
+      return Object.assign({}, state, {
+        scaling: newScale
+      });
+    }
+
+    case SET_TRANSLATION:
```

Comment:
Out of curiosity, why are the switch-case statements now enclosed in `{}`? Usually I see this left out for simplicity -
 or is this the AirBnB style we're following? In any case, for consistency, `SET_TRANSLATION` is missing them brackets.

Reply:
The linter complains, but it is weird because it doesn't happen in every duck, and I couldn't figure out why linter was being so picky. I'm up for standardizing one way or another, but if we choose to remove the, we should update the linter so it won't complain. https://eslint.org/docs/rules/no-case-declarations

Reply:
Here's how I've got switch/case formatting set up in PFE, from some code I was writing earlier.
```javascript
handleRadioKeyDown(questionId, answerId, e) {
    switch (e.which) {
      case BACKSPACE:
      case SPACE:
        if (e.target.checked) {
          e.preventDefault();
          this.handleRadioToggle(questionId, answerId, e);
        }
        break;
      default:
    }
  }
```

Reply:
I think this is the corresponding line from `.eslintrc`.
`"indent": ["warn", 2, {"SwitchCase": 1}],`